core/controller/src/main/scala/whisk/core/controller/Controller.scala

```diff
@@ -202,7 +202,7 @@ object Controller {
 
     // if deploying multiple instances (scale out), must pass the instance number as the
     require(args.length >= 1, "controller instance required")
-    val instance = args(0).toInt
+    val instance = StringInstanceId(args(0).trim)
```

Comment:
since there's a need to validate the string (to be a legal topic name), can we `require` the regex matches on construction instead of failing later on `topicName`?

Reply:
Sure, it just seemed more clear to make it fail by the code that cares about it most. I can move it here though.

Reply:
Actually, cannot do it on construction unless we restrict to some specific prefix length - some instance id values will work with certain topic prefix, depending on prefix length. We could only reserve some fixed size for prefix, say 10 chars, and on construction require that id is < than MAX_LENGTH - 10?

Reply:
that's reasonable to me - i think we can be prescriptive about the string ids (regex, length). 
would that be adequate for your purpose?

Reply:
Regex validation? may be problematic. But max length < 230 validation would be fine. (currently the regex doesn't validate, per se, just replaces invalid-for-kafka chars)